build/downloader/src/cli.ts

```diff
@@ -132,11 +171,13 @@ export const installFile = async (toolName: string, common: CommonArgs, toolVers
   } finally {
     downloadProgressBar.stop()
   }
+
+  await saveMetadata(release, common.appData)
 }
 
 export const useFile = async (toolName: string, version: string, common: CommonArgs) => {
   const toolFolder = path.resolve(common.appData, 'tools', toolName)
-  const underscoreVersion = version.replace(/\./g, '_')
+  const underscoreVersion = version.replace(/[\W_]+/g, '_')
```

Comment:
Please support prerelease tags as well as dev branches. This is fairly easy to do:

```ts
const parsed = semverParse(version)
if (!parsed) {
  return logger.error('Invalid semver version.')
}

const { major, minor, patch, prerelease } = parsed // prerelease is an array of string
const underscoreVersion = [major, minor, patch, ...prerelease].join('_')
```

Reply:
I don't want to validate the version, because in the pre-release, the version is the branch name. It should support rc out of the box

Reply:
Already works with the \W change

Reply:
Makes sense.